WordPress/UITests/README.md

```diff
@@ -85,17 +85,44 @@ Note that due to the mock server setup, tests cannot be run on physical devices
+* What screens are being tested (defined as screen objects in [Screens](https://github.com/wordpress-mobile/WordPress-iOS/tree/trunk/WordPress/UITestsFoundation/Screens)).
+* Whether there are repeated flows across tests (defined in [Flows](https://github.com/wordpress-mobile/WordPress-iOS/tree/trunk/WordPress/UITests/Flows)).
+* What network requests are made during the test (defined in [API-Mocks](https://github.com/wordpress-mobile/WordPress-iOS/tree/trunk/API-Mocks)).
 
-It's preferred to focus UI tests on entire user flows, and group tests with related flows or goals in the same test suite.
+Tests classes are grouped together in [Tests](https://github.com/wordpress-mobile/WordPress-iOS/tree/trunk/WordPress/UITests/Tests)
 
 When you add a new test, you may need to add new screens, methods, and flows. We use page objects and method chaining for clarity in our tests. Wherever possible, use an existing `accessibilityIdentifier` (or add one to the app) instead of a string to select a UI element on the screen. This ensures tests can be run regardless of the device language.
 
+## Naming convention
+
+* When creating new tests, use this format for the name to make it easier to see what the test is doing: `testActionFeature` e.g. `testCreateScheduledPost()`
+* When creating new methods, use this format: `actionObject` e.g. `closePostSettings()`
+* For assert methods, use this format `verifyWhatToVerify` e.g. `verifyPostExists()`
+* Note that there’s a common global method `assertScreenIsLoaded()` that can be used to assert all screens
+
+## Passing hard-coded `Strings` | `Numbers` in tests
+
+There are some cases where we would need to pass hard-coded values in the test, this should happen on the Test level and not on the screen level (unless there’s a really good reason). 
+
+This is so methods are not limited to being used with a fixed value and remain flexible. In the case where those values change we would also be able to update only the test file(s) without making changes elsewhere.
+
 ## Adding or updating network mocks
 
 When you add a test (or when the app changes), the request definitions for WireMock need to be updated in `API-Mocks/`. You can read WireMock’s documentation [here](http://wiremock.org/docs/).
 
-If you are unsure what network requests need to be mocked for a test, an easy way to find out is to run the app through [Charles Proxy](https://www.charlesproxy.com/) and observe the required requests.
+If you are unsure what network requests need to be mocked for a test, an easy way to find out is to run the app through [Proxyman](https://proxyman.io/) or [Charles Proxy](https://www.charlesproxy.com/) and observe the required requests.
+
+Currently, the project does not apply strict mock matching criteria, this means that if there are unmatched requests that are not being used by the test itself, the test should still work although errors like this can be seen in the logs:
```

Comment:
1. From seeing the `... in the logs:` it looks like you've intended to provide an error sample.
2. Maybe it's just me, but this paragraph might indirectly encourage to not keep the "not directly related" mocks in shape 🤔 Should it really be there, and does it make the overall approach better or cleaner?

Reply:
> From seeing the ... in the logs: it looks like you've intended to provide an error sample.

oh yes, you're right, the screenshot didn't get copied when i copied the rest of the doc 😓 i'll add that in. 

> this paragraph might indirectly encourage to not keep the "not directly related" mocks in shape

the reason i added that is that there are existing mock calls that are still not matched (not many _but_ there are) that will appear in the logs, so that would serve as a statement as to why it's happening and reduces the potential of having contributors ask about it. considering the endpoints are not used by the test, i would say that the approach doesn't impact the test itself but does clutter the console logs with "Request was not matched" logs. 

i personally think we should keep this explanatory statement. i also added a recommendation so it doesn't look like we're encouraging keeping the mock requests unmatched in https://github.com/wordpress-mobile/WordPress-iOS/pull/21594/commits/1c3cbadd9694413e88a9083e716ac361eb4e201f can you take another look and see if it reads better now?

Reply:
Yes, it's better now, thank you, Jos 🙇 (I'll provide my further comments all at once as a part of review shortly, sorry for cluttering the PR with separate standalone ones)